### DISCUSSION TITLE
Just bunch of big features, that i personally would want in-game

### DISCUSSION THREAD

**Discussion by Abkim27** [FakeFishGames/Barotrauma#10549] [Ideas and suggestions]
Reloading system

I think i have found a way to make "animation" for reloading. To make that "animation" need to write a code for:

1.Holding a weapon and pressing a reload key would make weapon one-handed and take out magazine to your left/right hand. Double-tapping reload key(or make a new key for that) would drop magazine immediately.

2.Left/Right hand with empty magazine would fall down, like if you store it in inventory. After storing mag somewhere, you take new magazine in left/right hand. New(AKA non-empty) magazine needs to be close to a gun when you don't aim and must have aim thing, so when you aiming weapon your magazine would be close to position of weapon.

3.Pressing reload key when weapon is not loaded would load magazine from left/right hand. If there is no magazine in hands, just don't do anything.

Also we can change reload speed for that, so when you have more weapon skill, you take out/load in weapon magazine faster.

New Inventory system

It would be hard to implement that in barotrauma, but making a new inventory system would be great. My opinion is to make Escape-from-Tarkov-like inventory, because it's a great example of inventory system. You can sit more than a hour just to fit everything from loot to your storage(and it wouldn't be boring)!

Water forces

We have a big problem with water forces. You can't get in ballast, if there is very big hole there, still can't climb ladder when water is pumped out. So, water forces needs to be weaker and not stun you at all if you wear safety harnees, exosuit or have a talent for that.

Grid

Most of builder games made by using grid, so why not to make barotrauma one of those? There needs to be new parameter for items - how much it takes cells in horizontal and how much it takes cells in vertical. Also every item would be snapped to a grid(not like how grid snapping works now! item's corners would be on grid's lines). This would make possible: customizing subs in-game(in campaign), new mechanics for upgrading items, by placing items near, pipes. Also sub building would be far more simple, but also decrease possibilities in decorating.

New item upgrade system

Talent trees, but for items! New window or tab in fabricator for upgrading items, there we would have trees for upgrading your items. As a talent trees, multple choices in one upgrade will lock other. Maybe there would be upgrade, that makes your plasma cutter work from battery, make welding tool much like ancient weapon and repair walls very fast at cost of using welding fuel very fast and etc. Suit upgrades will be as multple upgrade trees, one for each type of suit, so you will upgrade your diving suit to combat suit and with talents you could make upgrade to pucs or slipsuit. And yes, after upgrade sprite of an item would be changed.

**Comment by mygamingaccount** [comment]
these should be separate threads
